pkg/controller/podgc/gc_controller.go

```diff
@@ -59,14 +61,20 @@ func NewPodGC(kubeClient clientset.Interface, rpClients map[string]clientset.Int
 	}
 	gcc := &PodGCController{
 		kubeClient:              kubeClient,
-		resourceProviderClients: rpClients,
 		terminatedPodThreshold:  terminatedPodThreshold,
 		deletePod: func(tenant, namespace, name string) error {
 			klog.Infof("PodGC is force deleting Pod: %v/%v/%v", tenant, namespace, name)
 			return kubeClient.CoreV1().PodsWithMultiTenancy(namespace, tenant).Delete(name, metav1.NewDeleteOptions(0))
 		},
 	}
 
+	// key "0" is special case for TP client itself
+	// todo: avoid using magic literal "0"
+	gcc.kubeClientForNodes = map[string]clientset.Interface{"0": kubeClient}
```

Comment:
Checked scale up and scale out log again. Scale out has following log:
I0324 02:06:19.186483       1 gc_controller.go:195] Found orphaned Pod system/kube-system/etcd-server-events-poc430-032321-2tp2rpx500-kubemark-tp-1-master assigned to the Node poc430-032321-2tp2rpx500-kubemark-tp-1-master. Deleting.

But scale up does not have any.
sonya-uswest2:/home/sonyali/logs/perf-test/gce-500/arktos/up-poc430-032321-500/logs/kubemark_master/log

I suspect that in case of scale up, resourceProviderClients will take kubeClient as its value. That's why I this log could cause double kubeclients to node query.

Reply:
looked at the relevant code, for scale up (not scale out), gcc is created w/ this param as nil map - which implies double putting is not an issue; for scale out, this param passed in as map of RP clients, which disregards double putting too.

Reply:
Turns out there seems an existing defect in the current code which cannot list nodes for scale-up case.
 
Looked at sonya-uswest2:/home/sonyali/logs/perf-test/gce-500/arktos/up-poc430-032321-500/logs/kubemark_master/log/kube-controller-manager.log, there were a lot of error message
```
E0324 02:25:09.522145       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:25:29.522666       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:25:49.523044       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:26:09.523463       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:26:29.523887       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:26:49.524245       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:27:09.524656       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:27:29.525092       1 gc_controller.go:177] Error listing nodes from all resource partition. err: map[]
E0324 02:27:49.525469     ...
```

WRT this PR, it seems double-dipping out of question. Actually, it seems able to fix the aforementioned defect.

Reply:
Thanks for checking this out